src/charwise.rs

```diff
@@ -66,7 +66,7 @@ use crate::{MatchKind, Output};
 // The maximum BASE value used as an invalid value.
 pub(crate) const BASE_INVALID: i32 = i32::MAX;
 // The maximum output position value used as an invalid value.
-pub(crate) const OUTPUT_POS_INVALID: u32 = u32::MAX;
+pub(crate) const OUTPUT_POS_INVALID: u32 = u32::MAX >> 8;
```

Comment:
The fixing is not needed in the current version.

Reply:
I guess this fix is needed to check whether output value is valid in `output_pos()`, and I think it looks better to fix this since output pos requires 3 bytes in the new layout. Or, should I code this with keeping `OUTPUT_POS_INVALID` unchanged like the following? 

```rust
pub fn output_pos(&self) -> Option<u32> {
        Some(self.opos_ch | CHECK_MASK).filter(|&x| x != OUTPUT_POS_INVALID) 
}
```

Reply:
The state layouts in bytewise and charwise are distinct:
 - https://github.com/daac-tools/daachorse/blob/change-layout-state/src/lib.rs#L210
 - https://github.com/daac-tools/daachorse/blob/change-layout-state/src/charwise.rs#L943

Your new modification is not related to the charwise implementation.

Reply:
Oh, I forgot that `OUTPUT_POS_INVALID` is appeared in two files. Yes, the fix of `charwise.rs` is not needed.

Reply:
I found the simple fix to modify `OUTPUT_POS_INVALID` only in `lib.rs` not in `charwise.rs` does not work. Since some codes assume they are the same value. For example, `nfa_builder.rs` import `OUTPUT_POS_INVALID` in `lib.rs`, but `lib.rs` and `charwise.rs` import `nfa_builder.rs`. This assumption causes a problem.

Reply:
My solution is to 
 1. define `OUTPUT_POS_INVALID = u32::MAX` in `nfa_builder.rs` independently, and
 2. add to check if `NfaBuilderState::output_pos == nfa_builder::OUTPUT_POS_INVALID` in `builder.rs` of both bytewise and charwise.

For example, in construction of bytewise daachorse, [this line](https://github.com/daac-tools/daachorse/blob/change-layout-state/src/builder.rs#L359) should be modified into

```rust
if s.output_pos == nfa_builder::OUTPUT_POS_INVALID {
    self.states[idx].set_output_pos(crate::OUTPUT_POS_INVALID);
} else {
   self.states[idx].set_output_pos(s.output_pos);
}
```

`charwise/builder.rs` is as well.

Reply:
Since using `Option<NonZeroU32>` needs to change several parts of codes, it would be better we use @kampersanda 's solution at this time. And then modify to use `Option<NonZeroU32>` in another PR if needed.
updated in https://github.com/daac-tools/daachorse/pull/32/commits/4847bd884b0b557388768d35ba53c115596490c0.

Reply:
@kg86 Could you modify the following part as well as `DoubleArrayAhoCorasickBuilder`,
https://github.com/daac-tools/daachorse/blob/change-layout-state/src/charwise/builder.rs#L262
because `nfa_builder::OUTPUT_POS_INVALID` is independent from `charwise::OUTPUT_POS_INVALID`.

Reply:
Thanks for the pointing out.
updated in https://github.com/daac-tools/daachorse/pull/32/commits/da955e7f6e7a43ed427c92cdb775bb1aa6c86d18.